src/certificates.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manage the TLS Certificates."""
+
+import logging
+import os
+import pwd
+from pathlib import Path
+from typing import Sequence
+
+from charms.tls_certificates_interface.v4.tls_certificates import (
+    CertificateRequestAttributes,
+    PrivateKey,
+    ProviderCertificate,
+    TLSCertificatesRequiresV4,
+)
+
+from errors import (
+    TLSCertificateFileError,
+    TLSCertificateIntegrationNotExistError,
+    TLSCertificateNotAvailableError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def load_certificates(
+    common_names: Sequence[str],
+    username: str,
+    certificates_path: Path,
+    certificates: TLSCertificatesRequiresV4,
+) -> dict[str, Path]:
+    """Store the certificates available.
+
+    Args:
+        common_names: The common name of the certificates to check if available and store.
+        username: The name of the user to own the certificate file.
+        certificates_path: The directory to store the certificates.
+        certificates: The TLSCertificateRequiresV4 object.
+
+    Raises:
+        TLSCertificateIntegrationNotExistError: Unable to found the tls-certificate integration.
+        TLSCertificateNotAvailableError: At least one certificate is not available.
+
+    Returns:
+        The mapping of certificate common name to the file path of the corresponding
+        certificate. If the certificate is not available the entry will not exist in the
+        mapping.
+    """
+    relation = certificates.charm.model.get_relation(certificates.relationship_name)
+    if not relation:
+        raise TLSCertificateIntegrationNotExistError("TLS certificate integration not found")
+
+    logger.info("Loading the certificate available over tls-certificates integration")
+
+    common_name_to_cert = {}
+    for request in [CertificateRequestAttributes(common_name=name) for name in common_names]:
```

Comment:
You can use a generator expression here (switch the `[]` to `()`) since the list is not being used anyway